velox/core/Expressions.h

```diff
@@ -368,6 +371,9 @@ class ConcatTypedExpr : public ITypedExpr {
     if (!casted) {
       return false;
```

Comment:
wait looks like the problem is in comparing the inpits
func(z) and func(b) should always be the same 
if z==b regardless of type of func(z) and type of func(b)

the problem could be in comparing field references

Reply:
Discussed offline.